pkg/pillar/hypervisor/containerd.go

```diff
@@ -50,17 +49,18 @@ func (ctx ctrdContext) Task(status *types.DomainStatus) types.Task {
 func (ctx ctrdContext) Setup(status types.DomainStatus, config types.DomainConfig, aa *types.AssignableAdapters, file *os.File) error {
 	diskStatusList := status.DiskStatusList
 	domainName := status.DomainName
+
+	if !status.IsContainer {
+		return logError("failed to run domain %s: not based on an OCI image", domainName)
+	}
+
 	spec, err := containerd.NewOciSpec(domainName)
 	if err != nil {
 		return logError("requesting default OCI spec for domain %s failed %v", domainName, err)
 	}
-
-	if len(diskStatusList) > 0 && diskStatusList[0].Format == zconfig.Format_CONTAINER {
-		if err := spec.UpdateFromVolume(diskStatusList[0].FileLocation); err != nil {
-			return logError("failed to update OCI spec from volume %s (%v)", diskStatusList[0].FileLocation, err)
-		}
+	if err := spec.UpdateFromVolume(diskStatusList[0].FileLocation); err != nil {
```

Comment:
Nit but shouldn't we check for len(distStatusList) > 0 before accessing it as in the old code?

Reply:
Or an explicit log.Fatal check if diskStatusList is empty somewhere earlier?

Reply:
I think log.Fatal is a good idea. The rationale here is that it can't happen (but it is nice to check anyway) since the fact that something can be run as an OCI container is now predicated on this very check being moved to the domainConfig creation time. I'll add log.Fatal.